lib/oauth.js

```diff
@@ -378,7 +379,7 @@ internals.Client.prototype._request = function (method, uri, params, oauth, opti
     oauth.oauth_nonce = Cryptiles.randomString(internals.nonceLength);
     oauth.oauth_timestamp = Math.floor(Date.now() / 1000).toString();
     oauth.oauth_consumer_key = this.settings.clientId;
-    oauth.oauth_signature_method = 'HMAC-SHA1';
+    oauth.oauth_signature_method = this.settings.signatureMethod || 'HMAC-SHA1';
```

Comment:
If you change the Joi validation, then this is not needed anymore and you can just set `this.settings.signatureMethod`